From a241a7bef2fb28231ff2ec4a5e1dc9e4467c6a5b Mon Sep 17 00:00:00 2001
From: Selvavignesh Shanmugam <sshanmugam@maxlinear.com>
Date: Mon, 26 Aug 2024 16:24:13 +0800
Subject: [PATCH] WLANRTSYS-84263 Add STA entry Null check in send_assoc_resp

Issue:
If the MLO assoc translation occurs, STA entry will be Null
while handling the assoc response. Since, there is no null
check for the STA, segfault occurs.

Fix:
Add Null check for the STA entry.

FIXUP to WLANRTSYS-62845 AP MLD Auth Assoc and eapol changes
---
 src/ap/ieee802_11.c | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/src/ap/ieee802_11.c b/src/ap/ieee802_11.c
index 60c9e444e..59909d42b 100644
--- a/src/ap/ieee802_11.c
+++ b/src/ap/ieee802_11.c
@@ -5374,10 +5374,12 @@ rsnxe_done:
 		p = hostapd_eid_wmm(hapd, p);
 
 #ifdef CONFIG_IEEE80211BE_MXL_MLO
-	p = mxl_assoc_add_ml_ie(hapd, sta, p, reply, sta->mxl_sta_info.sta_profile_status);
-	if (!p) {
-		res = WLAN_STATUS_UNSPECIFIED_FAILURE;
-		goto done;
+	if (sta) {
+		p = mxl_assoc_add_ml_ie(hapd, sta, p, reply, sta->mxl_sta_info.sta_profile_status);
+		if (!p) {
+			res = WLAN_STATUS_UNSPECIFIED_FAILURE;
+			goto done;
+		}
 	}
 #endif /* CONFIG_IEEE80211BE_MXL_MLO */
 
-- 
2.43.0

